server/modules/record/repository/nodeRepository.js

```diff
@@ -169,26 +176,6 @@ export const deleteNodesByNodeDefUuids = async (surveyId, nodeDefUuids, client =
     dbTransformCallback
   )
 
-const stringifyValue = value => {
-  const isJsonString = str => {
-    try {
-      return R.is(Object, JSON.parse(str))
-    } catch (e) {
-      return false
-    }
-  }
-
-  return R.ifElse(
-    R.isNil,
-    R.identity,
-    R.ifElse(
-      v => R.is(String, v) && isJsonString(v),
-      R.identity,
-      v => JSON.stringify(v)
-    )
-  )(value)
-}
-
```

Comment:
I think there is a reason why this function was here... have you tested the insert/update of all different type of nodes?

Reply:
The test suite runs, and I was hoping to find if there was some rationale for this.
NULLs can be encoded as JSON nulls without any difference, but the double-encoding prevention seems strange.
Something feels very wrong if an input, what was supposed to be just a string, `"{}"` is automatically collapsed to a JSON object `{}`?

Reply:
Yes, it needs to be improved but I don't think it will work for all node types, can you check it please and postpone this improvement to better times?

Reply:
actually this method can be removed if using JSON.stringify in conjunction with cast to jsonb.
i.e. line 69 becomes ```VALUES ($1, $2, $3, $4, $5::jsonb, $6::jsonb)```
and line 133 becomes ``` SET value = $1::jsonb,```